Automattic/jetpack
# wpcomsh: Fix PHP 8 fatal on managed plugins

In some obscure scenarios, a plugin update check would trigger a PHP fatal on PHP 8. Reported in Automattic/jpop-issues#9074.

The fix is simple: cast the `get_option()` return value to an int.

I checked for other instances of math operations next to `get_option()` and fixed the three others I found.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






While the fix is simple, I struggled to come up with a way to reproduce this consistently. I was able to confirm the error on one of my test sites after poking at it for some time, but the trigger is unclear. What I do know is:

* One needs to have `XMLRPC_REQUEST` defined.
* One needs to have a `auto_updater.lock` option set to a non-numeric value (e.g. an empty string).
* The plugin update mechanism needs to trigger.

For the latter point, though, there are all sorts of preventative conditions in core (and probably elsewhere), so I don't have testing steps.

Instead, just look at the code and you'll see what it does.
